File: DroneStation.cpp
/**
@file
@date 2017/07/21
@author Minji and hexoul
@brief
*/
 
#include "DroneStation.h"
#include <tuple>
DroneStation::DroneStation(int _nMaxDrone, double _coverRange, double _stationLng, double _stationLat)
{
	nMaxDrone = _nMaxDrone;
	coverRange = _coverRange;
	stationLng = _stationLng;
	stationLat = _stationLat;
}

void DroneStation::setDroneNum(int n)
{
	if(n < 0 || n > nMaxDrone) return;
	nDrone = n;
}


/**
@brief upgrade the number of drone in the drone station
@details upgrade the change of the number of drone because of event right before
@param
@return
*/
void DroneStation::updateChargingDrones(Time currentTime)
{
	for(auto it =drones.begin() ; it != drones.end() ;it++)
	{
		if(it->returnStatus()== 2) 
		{
			it->charge(it->getChargeStartTime(), currentTime); //battery charged from centerArrivalTime

			if(it->returnBattery() >= 100)
			{
				it->setBattery(100);
				it->setStatus(0);
			}
		}
	}
}

File: Module/DroneStation.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DroneTransferSimulator
{
    public class DroneStation
    {
        public List<Drone> drones = new List<Drone>();
        public double coverRange;
        public double stationLng, stationLat;
        public string name;
        public Address addr;    

        public DroneStation(string _name, double _stationLat, double _stationLng, double _coverRange, Address _addr)
        {
            name = _name;
            coverRange = _coverRange;
            stationLat = _stationLat;
            stationLng = _stationLng;
            addr = _addr;
        }

        public DroneStation(string _name, double _stationLat, double _stationLng, double _coverRange)
        {
            name = _name;
            coverRange = _coverRange;
            stationLat = _stationLat;
            stationLng = _stationLng;
        }

        public void updateChargingDrones(DateTime currentTime)
        {
            foreach(Drone droneElement in drones)
            {
                if(droneElement.returnStatus() == Drone.droneType.D_CHARGING)
                {
                    droneElement.charge(droneElement.getChargeStartTime(), currentTime); // battery charged from centerArrivalTime

                    if(droneElement.returnBattery() >= 100)
                    {
                        droneElement.setBattery(100);
                        droneElement.setStatus(Drone.droneType.D_IN_STATION);
                    }
                }
            }
        }

        public Address getStationAddress()
        {
            return addr;
        }

        public void setStationAddress(Address _addr)
        {
            addr = _addr;
        }
    }
}
